### DISCUSSION TITLE
tutorials/voyage-part1/04-building-out-the-subgraphs

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#77] [course feedback]
# tutorials/voyage-part1/04-building-out-the-subgraphs

Learn what makes a subgraph different from a regular GraphQL server, and set up two subgraph servers using @apollo/subgraph

https://www.apollographql.com/tutorials/voyage-part1/04-building-out-the-subgraphs

**Comment by meppiel** [comment]
“Drag ‘n’ Drop” doesn’t work on an iPad. :)

**Reply by torressam333** [reply]
Curious if you are coding along on your ipad too?

**Comment by nijhoomroy** [comment]
Is there a typescript version of this repo to code with?

**Reply by lizhennessy** [reply]
Hi there @nijhoomroy ! We don't have a TypeScript version of this particular course, however we do have a separate Federation course that uses TypeScript! Here's the link: https://www.apollographql.com/tutorials/federation-typescript

**Reply by nijhoomroy** [reply]
Thank you! Does this cover everything from here?

**Reply by lizhennessy** [reply]
@nijhoomroy yes! We cover the same concepts about Rover, the router, and entities. :) Let me know if you have any other questions.

**Comment by aashaykadakia** [comment]
The directive 'link' should be defined as repeatable if its repeated on an SDL element
Unknown directive "link"

When i try to extend schema, this error pops up.

**Comment by otlyn** [comment]
Great series! Loving it.
